pkg/es/client/index_client.go

```diff
@@ -41,6 +44,32 @@ type IndicesClient struct {
 	MasterTimeoutSeconds int
 }
 
+// Create the indices only client using config.Configuration
+func CreateIndicesClient(c *config.Configuration, logger *zap.Logger) (*IndicesClient, error) {
```

Comment:
I don't follow why we need this function. The pattern in es.Factory is to use config.NewClient, which takes configuration.